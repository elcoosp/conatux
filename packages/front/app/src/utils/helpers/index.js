export const pipe = {
  execAsync: (...funcs) => context =>
    funcs.reduce(async (ctx, func) => {
      await func(ctx)
      return ctx
    }, context)
}
