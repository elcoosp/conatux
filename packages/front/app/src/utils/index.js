export { default as testing } from './testing'
export { default as linkify } from './linkify'
