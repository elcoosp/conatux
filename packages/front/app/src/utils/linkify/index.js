import React from 'react'
import { Link } from 'react-router-dom'

export const labelize = (path, label) => ({
  path,
  label
})

export const fromRoutes = (routesTree, componentName) => {
  if (['label', 'path'].every(prop => routesTree.hasOwnProperty(prop))) {
    const { path, label } = routesTree
    const link = props => (
      <Link to={path} {...props}>
        {label}
      </Link>
    )
    link.displayName = componentName
    return link
  }

  return Object.entries(routesTree).reduce(
    (links, [routeName, nestedRoute]) => {
      const componentName =
        routeName.slice(0, 1).toUpperCase() + routeName.slice(1)
      links[componentName] = fromRoutes(nestedRoute, componentName)
      return links
    },
    {}
  )
}
