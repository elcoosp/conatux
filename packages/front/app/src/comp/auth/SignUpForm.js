import React from 'react'
import * as yup from 'yup'
import { LabelledField, Form } from '@:/comp/helpers/form'

// Use spected
const validationSchema = yup.object().shape({
  username: yup
    .string()
    .required()
    .min(4)
    .max(12),
  email: yup
    .string()
    .email()
    .max(40)
    .required()
})

const initialValues = {
  email: '',
  password: '',
  passwordConfirmation: '',
  username: ''
}

const SignUpForm = () => (
  <Form
    initialValues={initialValues}
    validationSchema={validationSchema}
    // onSubmit={(values, { setSubmitting }) => {}}
  >
    <LabelledField label="Username" type="text" name="username" />
    <LabelledField label="Email" type="email" name="email" />
    <LabelledField label="Password" type="password" name="password" />
    <LabelledField
      label="Password confirmation"
      type="password"
      name="passwordConfirmation"
    />
  </Form>
)

export default SignUpForm
