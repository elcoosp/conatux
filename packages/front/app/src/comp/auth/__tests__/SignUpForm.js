import { desc } from '@:/utils/testing'
import SignUpForm from '../SignUpForm'

const props = {}

desc.compWithSnap({ SignUpForm }, props, () => {})
