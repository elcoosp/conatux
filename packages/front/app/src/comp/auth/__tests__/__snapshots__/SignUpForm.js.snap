// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> should match snapshot 1`] = `
<Form
  initialValues={
    Object {
      "email": "",
      "password": "",
      "passwordConfirmation": "",
      "username": "",
    }
  }
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": false,
      "_nodes": Array [
        "email",
        "username",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "required": false,
            "undefined": false,
          },
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "username": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "max": true,
            "min": true,
            "required": false,
          },
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <LabelledField
    label="Username"
    name="username"
    type="text"
  />
  <LabelledField
    label="Email"
    name="email"
    type="email"
  />
  <LabelledField
    label="Password"
    name="password"
    type="password"
  />
  <LabelledField
    label="Password confirmation"
    name="passwordConfirmation"
    type="password"
  />
</Form>
`;
